libnetdata/url/url.c

```diff
@@ -52,17 +52,73 @@ char *url_decode(char *str) {
     return url_decode_r(buf, str, size);
 }
 
+//decode %XX character or return 0 if cannot
+char url_percent_escape_decode(char *s) {
+    if(likely(s[1] && s[2]))
+        return from_hex(s[1]) << 4 | from_hex(s[2]);
+    return 0;
+}
+
+//this (utf8 string related) should be moved in separate file in future
+char url_utf8_get_byte_length(char c) {
+    char length = 0;
+    while(likely(c & 0x80)) {
+        length++;
+        c <<= 1;
+    }
+    //4 byte is max size for UTF-8 char
+    //10XX XXXX is not valid character -> check length == 1
+    if(length > 4 || length == 1)
+        return -1;
+    //0XXX XXXX means 1 byte
+    return (length) ? length : 1;
+}
+
+//decode % encoded UTF-8 characters and copy them to *d
+//return count of bytes written to *d
+char url_decode_multibyte_utf8(char *s, char *d, char* d_end) {
```

Comment:
> return count of bytes written to *d

> return byte_length;

it looks like it returns `byte_length`

`s` => `src`, `d` => `dst`, return `bytes_written` - after it you don't need comments.

I think you don't need comments just to explain naiming.

Maybe i am missing something and this is not `c` way. What do you think?

Reply:
ok, since this reflects surrounded code i have no question. But i strongly believe this is not part of the style we accepted.